liquibase-core/src/test/groovy/liquibase/LiquibaseTest.groovy

```diff
@@ -93,8 +94,19 @@ class LiquibaseTest extends Specification {
 
         mockHubService = (MockHubService) Scope.currentScope.getSingleton(HubServiceFactory).getService()
         mockHubService.reset()
+        def databaseChangeLog = new DatabaseChangeLog(changeLogId: MockHubService.randomUUID.toString())
+        databaseChangeLog.addChangeSet(new ChangeSet(
```

Comment:
Hi @RichardBradley, if I'm not misunderstanding this part of the code. Are we not missing the `isUpToDateFastCheck()` here? If not could you please clarify to me how this test works? 

Thanks,
Daniel.

Reply:
(This comment is based on my recollection only; I haven't re-run this code since I made this PR in Nov 2021.)

This change is not directly relevant to this PR.

When working on this PR, I found that this test was failing already. The failure was in unrelated code when constructing a changelog with no changesets in it. I needed to add an empty changeset to the changelog in order to fix the test.

> Are we not missing the isUpToDateFastCheck() here? 

No, client code should not directly invoke "isUpToDateFastCheck", but should always call the public methods on Liquibase such as update()

>  If not could you please clarify to me how this test works?

I'm not exactly sure what this test is attempting to verify. It fails before this change and passes after it, so I thought it was an improvement.

Reply:
Ohh I see, I thought this test was intended to test this change (cc: @XDelphiGrl ). Thanks for clarifying and thank you for your contributions on this PR. 

Daniel.

Reply:
Thanks for the response, @RichardBradley - fixing broken tests is always a good thing!

@MalloD12, can we add a test for `isUpToDateFastCheck`?

Reply:
Yeap,  we can do that!

Reply:
I've added two unit test to validate `upToDateFastCheck()` method. One scenario to validate a context-label entry does not exists and one to validate it does.

Could you @XDelphiGrl  please review this when you have a chance?

Thanks,
Daniel.